drivers/sensor/qdec_nxp_s32/qdec_nxp_s32.c

```diff
@@ -0,0 +1,424 @@
+	LISTIFY(__DEBRACKET eMIOS_INSTANCE_COUNT, EMIOS_NXP_S32_INSTANCE_CHECK, (|), node_id)
+
+/* LCU Logic Input Configuration */
+#define LogicInputCfg_Common(mux_sel)								   \
+	{											   \
+		.MuxSel = mux_sel,								   \
+		.SwSynMode = LCU_IP_SW_SYNC_IMMEDIATE,						   \
+		.SwValue = LCU_IP_SW_OVERRIDE_LOGIC_LOW,					   \
+	};
+#define LogicInput_Config_Common(n, hw_lc_input_id, logic_input_n_cfg)				   \
+	{											   \
+		.xLogicInputId = {								   \
+			.HwInstId = DT_INST_PROP(n, lcu_inst),					   \
+			.HwLcInputId = hw_lc_input_id,						   \
+		},										   \
+		.pxLcInputConfig = &logic_input_n_cfg,						   \
+	};
+
+/* LCU Logic Output Configuration */
+#define LogicOutputCfg_Common(En_Debug_Mode, Lut_Control, Lut_Rise_Filt, Lut_Fall_Filt)		   \
+	{											   \
+		.EnDebugMode = (boolean)En_Debug_Mode,						   \
+		.LutControl = Lut_Control,							   \
+		.LutRiseFilt = Lut_Rise_Filt,							   \
+		.LutFallFilt = Lut_Fall_Filt,							   \
+		.EnLutDma = (boolean)FALSE,							   \
+		.EnForceDma = (boolean)FALSE,							   \
+		.EnLutInt = (boolean)FALSE,							   \
+		.EnForceInt = (boolean)FALSE,							   \
+		.InvertOutput = (boolean)FALSE,							   \
+		.ForceSignalSel = 0U,								   \
+		.ClearForceMode = LCU_IP_CLEAR_FORCE_SIGNAL_IMMEDIATE,				   \
+		.ForceSyncSel = LCU_IP_SYNC_SEL_INPUT0,						   \
+	};
+#define LogicOutput_Config_Common(n, logic_output_cfg, hw_lc_output_id)				   \
+	{											   \
+		.xLogicOutputId = {								   \
+			.HwInstId = DT_INST_PROP(n, lcu_inst),					   \
+			.HwLcOutputId = hw_lc_output_id,					   \
+			.IntCallback = NULL_PTR,						   \
+		},										   \
+		.pxLcOutputConfig = &logic_output_cfg,						   \
+	};
+
+#define LCU_IP_INIT_CONFIG(n)									\
+	const Lcu_Ip_LogicInputConfigType LogicInput0Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_IN_0)					\
+	const Lcu_Ip_LogicInputConfigType LogicInput1Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_IN_1)					\
+	const Lcu_Ip_LogicInputConfigType LogicInput2Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_OUT_0)					\
+	const Lcu_Ip_LogicInputConfigType LogicInput3Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_OUT_1)					\
+												\
+	const Lcu_Ip_LogicInputType LogicInput0_Config =					\
+		LogicInput_Config_Common(n, 0U, LogicInput0Cfg)					\
+	const Lcu_Ip_LogicInputType LogicInput1_Config =					\
+		LogicInput_Config_Common(n, 1U, LogicInput1Cfg)					\
+	const Lcu_Ip_LogicInputType LogicInput2_Config =					\
+		LogicInput_Config_Common(n, 2U, LogicInput2Cfg)					\
+	const Lcu_Ip_LogicInputType LogicInput3_Config =					\
+		LogicInput_Config_Common(n, 3U, LogicInput3Cfg)					\
+												\
+	const Lcu_Ip_LogicInputType								\
+		*const Lcu_Ip_ppxLogicInputArray_Config[LCU_IP_NOF_CFG_LOGIC_INPUTS] = {	\
+			&LogicInput0_Config,							\
+			&LogicInput1_Config,							\
+			&LogicInput2_Config,							\
+			&LogicInput3_Config,							\
+		};										\
```

Comment:
Will this work for multiple instances? I don't see the instance number `n` being used here

Reply:
Same reply as above !

Reply:
Fixed by the change made for the comment comment

Reply:
@MaureenHelm .. Any S32 based quadrature decoder will have 4 inputs to the LCU, LogicInput0_Config,1,2,3 depict those 4 inputs. 
Now I have made LCU Input Index as a configurable parameter based on device tree selection, there is a logic in hal_nxp which will determine based on this index as to which Logic Cell to be used inside this particular LCU instance. 
Now this makes all the input and output LCU parameters as configurable as per qdec instance

Reply:
This still looks like it won't work for multiple instances because the symbol names (e.g., `LogicInput0Cfg`) aren't unique for each instance. You'll end up with duplicate definitions of the same symbol.

Reply:
@sumitbatra-nxp, Maureen's point is valid here. You have defined the LCU_IP_INIT_CONFIG() macro that takes an instance. Below it is invoked via the DT__INST_FOREACH which implies you intend for this to work with multiple instances. So when LCU_IP_CONFIG() is invoked multiple times it will try to define the symbols multiple times. If this is indeed intended to support multiple instances, then a symbol like LogicInput0Cfg would need a _##n## to make the name unique.